js/util/locale-bundle-loader.js

```diff
@@ -0,0 +1,61 @@
+import { RipeCommonsPlugin, RipeCommonsCapability } from "../abstract";
+
+export class LocaleBundleLoaderPlugin extends RipeCommonsPlugin {
+    async load() {
+        await super.load();
+        this.ripeProvider = await this.owner.getPluginByCapability("ripe-provider");
+        this.loadedBundles = [];
+
+        this.onLocaleChangedWatch = this.owner.bind(
+            "locale_changed",
+            async locale => await this.onLocaleChanged(locale)
+        );
+    }
+
+    async unload() {
+        this.owner.unbind("locale_changed", this.onLocaleChangedWatch);
+        this.ripeProvider = null;
+        await super.unload();
+    }
+
+    async onLocaleChanged(locale = this.locale) {
+        this._tryLoadLocaleBundle(locale);
+    }
+
+    async _tryLoadLocaleBundle(locale) {
+        // if ripeProvider isn't ready, try again later
+        if (!this.ripeProvider.ripe) {
+            const ripeReadyInterval = setInterval(async () => {

```

Comment:
avoid busy waiting and use a promise to wait for an event triggered when ripe provider is ready